cv19ResSupportV3.Tests/V3/Gateways/ResidentGatewayTest.cs

```diff
@@ -5,89 +5,477 @@
 using FluentAssertions;
 using Microsoft.EntityFrameworkCore;
 using NUnit.Framework;
+using Bogus;
+using System.Linq;
 
 namespace cv19ResSupportV3.Tests.V3.Gateways
 {
     [TestFixture]
     public class ResidentGatewayTest : DatabaseTests
     {
         private ResidentGateway _classUnderTest;
+        private Faker _faker = new Faker();
 
         [SetUp]
         public void Setup()
         {
             _classUnderTest = new ResidentGateway(DatabaseContext);
         }
 
+        // not matching first name, but matching uprn or dob
+        // Does not throw an error if... empty db
+
+        #region Duplicate Resident Finding Tests (FindResident)
+
+        // If NHS numbers match, then:
+        // Resident id is returned as duplicate.
+
         [Test]
-        public void FindResidentWithUprnAndNameReturnsTheResidentIdIfItExists()
+        public void FindResidentReturnsAMatchIdWhenThereIsAResidentWithMatchingNHSNumber()
         {
-            var existingResident = new ResidentEntity { Uprn = "uprn", FirstName = "FirstName", LastName = "LastName" };
+            // arrange
+            var matchingNhsNumber = _faker.Random.Hash();
 
-            var findResidentCommand = new FindResident
+            var searchParameters = new FindResident
             {
-                Uprn = "uprn",
-                FirstName = "FirstName",
-                LastName = "LastName",
-                DobDay = "Day",
-                DobMonth = "Month",
-                DobYear = "Year"
+                NhsNumber = matchingNhsNumber
             };
 
-            DatabaseContext.ResidentEntities.Add(existingResident);
+            // control resident
+            var existingResidentNoNhsNumberMatch = new ResidentEntity
+            {
+                NhsNumber = _faker.Random.Hash()
+            };
+
+            var existingResidentNhsNumberMatch = new ResidentEntity
+            {
+                NhsNumber = matchingNhsNumber
+            };
+
+            DatabaseContext.ResidentEntities.Add(existingResidentNoNhsNumberMatch);
+            DatabaseContext.ResidentEntities.Add(existingResidentNhsNumberMatch);
             DatabaseContext.SaveChanges();
-            var response = _classUnderTest.FindResident(findResidentCommand);
-            var createdRecord = DatabaseContext.ResidentEntities.Find(response);
-            var expectedRecord = new ResidentEntity
+
+            var nhsNumberMatchResidentId = DatabaseContext.ResidentEntities.FirstOrDefault(r => r.NhsNumber == matchingNhsNumber)?.Id;
```

Comment:
Do we need this variable? Could we just match against ```existingResidentNhsNumberMatch.Id``` ? That way we don't rely on the database on the both sides of the assertion

Reply:
Unfortunately no. The Ids are auto-generated upon inserting a record, meaning we don't know what the id is until we insert the object, and then retrieve it to look at what the id is.